package/kedro_viz/integrations/kedro/hooks.py

```diff
@@ -0,0 +1,106 @@
+# pylint: disable=broad-exception-caught
+# pylint: disable=protected-access
+"""`kedro_viz.integrations.kedro.hooks` defines hooks to add additional
+functionalities for a kedro run."""
+
+import json as json_lib
+import logging
+import sys
+from collections import defaultdict
+from typing import Any
+
+import pandas as pd
+from kedro.framework.hooks import hook_impl
+from kedro.io import DataCatalog
+
+from kedro_viz.integrations.kedro.utils import stats_order
+from kedro_viz.models.utils import get_file_size
+
+try:
+    # kedro 0.18.11 onwards
+    from kedro.io import MemoryDataset
+except ImportError:  # pragma: no cover
+    # older versions
+    from kedro.io import MemoryDataSet as MemoryDataset
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetStatsHook:
+    """Class to collect dataset statistics during a kedro run
+    and save it to a JSON file. The class currently supports
+    (pd.DataFrame, list, dict and pd.core.series.Series) dataset instances"""
+
+    def __init__(self):
+        self._stats = defaultdict(dict)
+
+    @hook_impl
+    def after_catalog_created(self, catalog: DataCatalog):
+        """Hook to be invoked after a data catalog is created.
+        Use this hook and get the file_size for the dataset if it has filepath.
+
+        Args:
+            catalog: The catalog that was created.
+        """
+        try:
+            datasets = catalog._data_sets
+
+            for dataset_name, dataset in datasets.items():
+                if not isinstance(dataset, MemoryDataset):
+                    file_path = dataset._filepath  # noqa: no-member
```

Comment:
This is quite fragile, `_filepath` is not a API of dataset. In addition do we track DataFrame only or we are tracking more dataset?

Reply:
I think for now we will track only DataFrame

Reply:
Yes even Circle CI build fails as the `AbstractDataset` does not contain _filepath field. Do you think it is better to read the file size in `DataNodeMetadata` class ? I am already doing that - please check line 571-573 in `package/kedro_viz/models/flowchart.py`

Reply:
May need to catch up on this later. Top of my head I think we should limit the scope to just DataFrame and don't try to address the most generic case now. `_filepath` is a problem because a `SQLTableDataSet` is a pandas dataframe but it doesn't have a filepath (worse if it has one it is the path of the `.sql` file instead of the dataframe).

Reply:
okay, if that is the case, getting the file size without checking for extension here `(line 569 in package/kedro_viz/models/flowchart.py)` will be a problem ? So should we check for file extensions ?